crates/agglayer-storage/src/tests/regression_db_deserialization.rs

```diff
@@ -0,0 +1,2238 @@
+        let network_id = NetworkId::new(key.network_id);
+        network_ids.insert(network_id);
+        cert_per_network_map.insert((network_id, key.height), cert_id);
+    }
+    println!(
+        "   ✅ Reconstructed {} certificate mappings for {} networks",
+        cert_per_network_map.len(),
+        network_ids.len()
+    );
+
+    // 2. Read and validate CertificateHeaders
+    println!("\n2. Reading certificate_header_cf...");
+    let mut headers: BTreeMap<CertificateId, CertificateHeader> = BTreeMap::new();
+    for cert_id_result in state_db
+        .keys::<CertificateHeaderColumn>()
+        .expect("Failed to iterate")
+    {
+        let cert_id = cert_id_result.expect("Failed to read cert ID");
+        let header: CertificateHeader = state_db
+            .get::<CertificateHeaderColumn>(&cert_id)
+            .expect("Failed to read header")
+            .expect("Header not found");
+
+        // Validate header structure
+        assert_eq!(header.certificate_id, cert_id, "Certificate ID mismatch");
+        assert_eq!(header.status, CertificateStatus::Pending);
+        assert!(header.epoch_number.is_some(), "Epoch number should be set");
+        assert!(
+            header.certificate_index.is_some(),
+            "Certificate index should be set"
+        );
+
+        // Verify it matches the cert_per_network mapping
+        if let Some(mapped_cert_id) = cert_per_network_map.get(&(header.network_id, header.height))
+        {
+            assert_eq!(
+                *mapped_cert_id, cert_id,
+                "Certificate mapping mismatch for network {} height {}",
+                header.network_id, header.height
+            );
+        }
+
+        headers.insert(cert_id, header);
+    }
+    println!("   ✅ Reconstructed {} certificate headers", headers.len());
+    println!("   ✅ All headers have valid epoch/index and match network mappings");
+
+    // 3. Read and validate LatestSettledCertificatePerNetwork
+    println!("\n3. Reading latest_settled_certificate_per_network_cf...");
+    let mut latest_settled_certs: HashMap<NetworkId, SettledCertificate> = HashMap::new();
+    for network_id_result in state_db
+        .keys::<LatestSettledCertificatePerNetworkColumn>()
+        .expect("Failed to iterate")
+    {
+        let network_id = network_id_result.expect("Failed to read network ID");
+        let settled_cert: SettledCertificate = state_db
+            .get::<LatestSettledCertificatePerNetworkColumn>(&network_id)
+            .expect("Failed to read settled cert")
+            .expect("Settled cert not found");
+
+        // Validate that this certificate exists in headers
+        if let Some(header) = headers.get(&settled_cert.0) {
+            assert_eq!(
+                header.network_id, network_id,
+                "Settled cert network mismatch"
+            );
+            assert_eq!(
+                header.height, settled_cert.1,
+                "Settled cert height mismatch"
+            );
```

Comment:
This check does not have much to do with encoding / decoding as such.

Basically it checks kind of database consistency properties. Checks like this are useful in many contexts since it is expected to hold after pretty much every operation. Would be nice to separate the db consistency checks into a function so they can be strategically sprinkled throughout the tests, serving as a sort of force multiplier for many of the existing tests. (Or even in the code base as `debug_assert!`).

Reply:
I agree - we could do it in the follow up.